bin/git-updatepr

```diff
@@ -38,12 +38,22 @@ if [[ "$new_refspec" != "$base_refspec" ]]; then
   exit 1
 fi
 
-if ! git cat-file -e "$commit_to_update" 2> /dev/null; then
-  echo "error: invalid commit sha: '$commit_to_update'" >&2
+# Get the commit to update and the branch name
+if git show-ref --quiet --verify "refs/heads/$sha_or_branch_to_update"; then
+  branch_name="$sha_or_branch_to_update"
+  commit_to_update=$(git rev-parse --verify "$branch_name" 2>/dev/null)
+else
+  commit_to_update="$sha_or_branch_to_update"
+  branch_name="$(git pilebranchname "$commit_to_update")"
+fi
+
+# Check if we got a valid SHA
+if ! git cat-file -e "$commit_to_update" 2>/dev/null; then
```

Comment:
do we need this in the new case? or should `show-ref` cover that? just wondering because it might confuse future us that `commit_to_update` isn't a commit in the current "pile" which it is in the other codepath

Reply:
Yeah i think theres some unnecessary checks happening, i've cleaned it up to only do the checks it needs to do. Let me know if you were thinking of something else.